client/src/modules/fill_form/modals/fill_form.modals.html

```diff
@@ -0,0 +1,250 @@
+          <label class="control-label"> {{ formItem.label }} </label>
+          <span style="color: #cecece"><em> {{ formItem.hint }} </em></span>
+          <div>
+            <span ng-if="formItem.required">
+              <input name="{{ formItem.name }}" ng-model="FillFormModalCtrl.form[formItem.name]" autocomplete="off" ng-maxlength="90" class="form-control" required>
+            </span>
+            <span ng-if="!formItem.required">
+              <input name="{{ formItem.name }}" ng-model="FillFormModalCtrl.form[formItem.name]" autocomplete="off" ng-maxlength="90" class="form-control">
+            </span>
+            
+            <div class="help-block" ng-messages="FillFormForm.name.$error" ng-show="FillFormForm.$submitted">
+              <div ng-messages-include="modules/templates/messages.tmpl.html"></div>
+            </div>
+          </div>
+        </div>
+
+        <!-- number-->
+        <div ng-if="formItem.typeForm === 'number'" class="form-group" ng-class="{ 'has-error' : FillFormForm.$submitted && FillFormForm.{{ formItem.name }}.$invalid }">
+          <label class="control-label"> {{ formItem.label }} </label>
+          <span style="color: #cecece"><em> {{ formItem.hint }} </em></span>
+          <div>
+            <span ng-if="formItem.required">
+              <input name="{{ formItem.name }}" type="number" ng-model="FillFormModalCtrl.form[formItem.name]" autocomplete="off" ng-maxlength="90" class="form-control" required>
+            </span>
+            <span ng-if="!formItem.required">
+              <input name="{{ formItem.name }}" type="number" ng-model="FillFormModalCtrl.form[formItem.name]" autocomplete="off" ng-maxlength="90" class="form-control">
+            </span>
+            <div class="help-block" ng-messages="FillFormForm.name.$error" ng-show="FillFormForm.$submitted">
+              <div ng-messages-include="modules/templates/messages.tmpl.html"></div>
+            </div>
+          </div>
+        </div>
+
+        <!-- Date -->
+        <div ng-if="formItem.typeForm === 'date'">
+          <span style="color: #cecece"><em> {{ formItem.hint }} </em></span>
+          <bh-date-editor
+            date-value="FillFormModalCtrl.form[formItem.name]"
+            label="{{ formItem.label }}"
+            allow-future-date="true"
+            required="formItem.valueRequired"
+            on-change="FillFormModalCtrl.onDateChange(date, formItem.name)">
+          </bh-date-editor>
+        </div>
+
+        <!-- Calculation -->
+        <div ng-if="formItem.typeForm === 'calculation'">
+          <span style="background-color: {{ FillFormModalCtrl.dataCollector.color }}; color: black"><em> <strong>{{ formItem.label }}</strong> : {{ formItem.calculation }}
+          <h5 ng-if="FillFormModalCtrl.updateMode">:{{ FillFormModalCtrl.form[formItem.name] }} </h5> </em></span> 
+        </div>
+
+        <!-- TextArea -->
+        <div ng-if="formItem.typeForm === 'text_area'">
```

Comment:
Since you are doing a lot of `formItem.typeForm === "x"`, could you use an [`ngSwitch`](https://docs.angularjs.org/api/ng/directive/ngSwitch)?  It would probably make the code easier to read.